### PR TITLE
[@mysten/ledgerjs-hw-app-sui] add displayOnDevice setting for getPublicKey and do some general clean-up type work

### PR DESCRIPTION
**Pull request by williamrobertson13**
## Description 

This PR adds a `displayOnDevice` param to `getPublicKey` in the LedgerJS binding library and does some re-factoring in the library code.

## Test Plan 
- Manual testing (Ledger flow works and can verify address using new library method)

---
If your changes are not user-facing and not a breaking change, you can skip the following section. Otherwise, please indicate what changed, and then add to the Release Notes section as highlighted during the release process.

### Type of Change (Check all that apply)

- [X] user-visible impact
- [ ] breaking change for a client SDKs
- [ ] breaking change for FNs (FN binary must upgrade)
- [ ] breaking change for validators or node operators (must upgrade binaries)
- [ ] breaking change for on-chain data layout
- [ ] necessitate either a data wipe or data migration

### Release notes
N/A

### CODE REVIEW

File: apps/wallet/src/ui/app/components/menu/content/VerifyLedgerConnectionStatus.tsx [hunk 1]
```diff
@@ -37,10 +37,6 @@ export function VerifyLedgerConnectionStatus({
     );
     const timeoutIdRef = useRef<number>();
 
-    useEffect(() => {
```
Comment (williamrobertson13) [inline apps/wallet/src/ui/app/components/menu/content/VerifyLedgerConnectionStatus.tsx:40]: I read asymmetric effects were bad to do on Twitter from the React ppl 😨

File: sdk/ledgerjs-hw-app-sui/src/Sui.ts [hunk 2]
```diff
@@ -2,39 +2,307 @@
 // SPDX-License-Identifier: Apache-2.0
 
 import type Transport from '@ledgerhq/hw-transport';
-import { Common } from './Common';
-import type { SignTransactionResult, GetVersionResult } from './Common';
-
-export type { SignTransactionResult, GetVersionResult };
+import sha256 from 'fast-sha256';
```
Comment (williamrobertson13) [inline sdk/ledgerjs-hw-app-sui/src/Sui.ts:5]: All of this is pretty much copied + pasted from the updated `Common.js`

File: apps/wallet/src/ui/app/components/menu/content/VerifyLedgerConnectionStatus.tsx (hunk 1 above)
Comment (Jordan-Mysten) [reply apps/wallet/src/ui/app/components/menu/content/VerifyLedgerConnectionStatus.tsx:40]: Does this mean we never clear the timeout though?

File: sdk/ledgerjs-hw-app-sui/package.json [hunk 3]
```diff
@@ -1,6 +1,6 @@
 {
     "name": "@mysten/ledgerjs-hw-app-sui",
-    "version": "0.1.0",
+    "version": "0.2.0-beta.0",
```
Comment (Jordan-Mysten) [inline sdk/ledgerjs-hw-app-sui/package.json:3]: Don't change this

**Review by Jordan-Mysten — APPROVED** [review]

File: apps/wallet/src/ui/app/components/menu/content/VerifyLedgerConnectionStatus.tsx (hunk 1 above)
Comment (williamrobertson13) [reply apps/wallet/src/ui/app/components/menu/content/VerifyLedgerConnectionStatus.tsx:40]: yeah but I took Dan's word as god 😆 

https://twitter.com/dan_abramov/status/1648713546803412993

File: apps/wallet/src/ui/app/components/menu/content/VerifyLedgerConnectionStatus.tsx [hunk 4]
```diff
@@ -51,7 +47,8 @@ export function VerifyLedgerConnectionStatus({
                             const suiLedgerClient = await connectToLedger();
                             const publicKeyResult =
                                 await suiLedgerClient.getPublicKey(
```
Comment (williamrobertson13) [inline apps/wallet/src/ui/app/components/menu/content/VerifyLedgerConnectionStatus.tsx:49]: I need to add the loading state back, but I'll do this on Monday

**Pull request merged**